### PR TITLE
Add prominent words presence check to the linking component

### PR DESCRIPTION
**Pull request by nataliashitova**
## Summary

  * Package(s) involved: components

  * Should the change be included in the package changelog?
    * [ ] No
    * [ ] Yes
  * Should the change be included in one or more plugin changelogs?
    * [ ] No (This change is already included in the Premium through [the premium PR](https://github.com/Yoast/wordpress-seo-premium/pull/2449).
    * [ ] Free
    * [ ] Premium
    * [ ] Other (please specify)
  * Package changelog item (if applicable): In the internal linking suggestions component, adds a check for the presence of the prominent words to determine which of the feedback strings should be rendered. If no prominent words are determined for a text (and hence no suggestions), the old message "Once you add a bit more copy..." is rendered. If prominent words are determined, but still no suggestions could be found, a new message is rendered: "We could not find any relevant articles on your website that you could link your post to.".
  * Plugin changelog item (if applicable): 

## Relevant technical choices:

* See [the Premium PR](https://github.com/Yoast/wordpress-seo-premium/pull/2449) for more context.

## Test instructions

This PR can be tested by following these steps:

* See [the Premium PR](https://github.com/Yoast/wordpress-seo-premium/pull/2449).

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * Internal linking suggestions

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #https://github.com/Yoast/wordpress-seo-premium/issues/2442

### CODE REVIEW

File: packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js [hunk 1]
```diff
@@ -94,13 +95,27 @@ class LinkSuggestions extends React.Component {
...
 		} );
 
+		const noRelevantPostsMessage = __(
+			"We could not find any relevant articles on your website that you could link to from your post.", "yoast-components" );
+
 		const moreCopyMessage = __(
 			"Once you add a bit more copy, we'll give you a list of related " +
 			"content here to which you could link in your post.", "yoast-components" );
 
+		// If there is not enough text to calculate Prominent Words an "Add a bit more copy" message is returned.
```
Comment (maartenleenders) [inline packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js]:
```suggestion
        const renderMessage = isEqual( this.props.prominentWords, [] ) ? moreCopyMessage : noRelevantPostsMessage;
        
		// If there is not enough text to calculate Prominent Words an "Add a bit more copy" message is returned.
```

Comment (maartenleenders) [reply packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js]: I think this method avoids some duplicate writing in the return function.

File: packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js [hunk 2]
```diff
@@ -94,13 +95,27 @@ class LinkSuggestions extends React.Component {
...
+					<p>{ articleLink }</p>
+				</div>
+			);
+		}
+
+		// Otherwise we return a message that no relevant posts are found.
 		return (
 			<div>
-				<p>{ moreCopyMessage }</p>
+				<p>{ noRelevantPostsMessage }</p>
```
Comment (maartenleenders) [inline packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js]:
```suggestion
				<p>{ renderMessage }</p>
```

File: packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js [hunk 3]
```diff
@@ -94,13 +95,27 @@ class LinkSuggestions extends React.Component {
...
 
+		const noRelevantPostsMessage = __(
+			"We could not find any relevant articles on your website that you could link to from your post.", "yoast-components" );
+
 		const moreCopyMessage = __(
 			"Once you add a bit more copy, we'll give you a list of related " +
 			"content here to which you could link in your post.", "yoast-components" );
 
+		// If there is not enough text to calculate Prominent Words an "Add a bit more copy" message is returned.
+		if ( isEqual( this.props.prominentWords, [] ) ) {
```
Comment (maartenleenders) [inline packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js]: this if-loop can deleted if you implement the suggestion mentioned above.

File: packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js [hunk 4]
```diff
@@ -94,13 +95,27 @@ class LinkSuggestions extends React.Component {
 			},
 		} );
 
+		const noRelevantPostsMessage = __(
```
Comment (maartenleenders) [inline packages/yoast-components/composites/LinkSuggestions/LinkSuggestions.js]: This const is created every time this function runs. I think you can declare it outside of the class and just reference it the ternary statement I suggested below.

The same goes for `moreCopyMessage`

**Comment by maartenleenders** [comment]
Small optimisation suggestion. 🚧 

I'm not sure the suggestions above use tabs or spaces properly, so it's best to *not* press `Commit suggestion` button, but to write them manually in PHPstorm (_if_ you agree, obviously).

**Comment by nataliashitova** [comment]
@maartenleenders, you've got really nice suggestions! As agreed, please implement them and move the PR to acceptance. 😄

**Comment by manuelaugustin** [comment]
Acceptance 👍

**Pull request merged**